backend/src/main/java/hu/flowacademy/vajdasagbrand/service/ItemService.java

```diff
@@ -21,6 +26,20 @@ public Item createItem(Item item) throws ValidationException {
         return itemRepository.save(item);
     }
 
+    public Item deleteById(String id) throws ValidationException {
+        Optional<Item> item = itemRepository.findById(id);
```

Comment:
Ha írsz egy saját függvényt, pl. `findFirstByIdAndDeletedAtNotNull` néven, akkor az adatbázisra bízhatod a törölt állapot vizsgálatát is és ezt is vissza tud térni Optional-nel

Reply:
kiszerveztem és beletettem az orElseThrow-t is ebbe a metódusba